src/lib/operators/join_sort_merge.cpp

```diff
@@ -941,7 +941,17 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractJoinOperatorImpl {
         Segments segments;
         _add_output_segments(segments, _sort_merge_join.input_table_left(), _output_pos_lists_left[pos_list_id]);
         _add_output_segments(segments, _sort_merge_join.input_table_right(), _output_pos_lists_right[pos_list_id]);
-        output_chunks[pos_list_id] = std::make_shared<Chunk>(std::move(segments));
+        auto output_chunk = std::make_shared<Chunk>(std::move(segments));
+        if (_sort_merge_join._primary_predicate.predicate_condition == PredicateCondition::Equals &&
+            _secondary_join_predicates.empty()) {
+          const ColumnID first_join_column = _sort_merge_join._primary_predicate.column_ids.first;
+          const ColumnID second_join_column =
+              static_cast<ColumnID>(_sort_merge_join.input_table_left()->column_count() +
+                                    _sort_merge_join._primary_predicate.column_ids.second);
+          output_chunk->finalize();
+          output_chunk->set_value_clustered_by({first_join_column, second_join_column});
```

Comment:
Why don't you set the sort flag here as well?

Reply:
currently, ordered_by is a flag that can be set only on one column. however since after the equi join, the join column appears two times in the result and we do not know, which one will be discarded later, it's hard to decide, which one should be labeled as sorted. do you think, that allowing multiple columns to be sorted at the same time would be a good idea here?

Reply:
Conceptually, I think it's totally fine to allow a chunk to be sorted by multiple columns at the same time, even though that might sound strange at first.
I doubt we will have such cases soon, but if we ever let Hyrise know about something like functional dependencies, this could also result in a table where we know for sure that a table is sorted by A **and** B (because it's sorted by A and B is in some form dependent of the values in A).